src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkUtils.java

```diff
@@ -194,7 +195,9 @@ private static void createDataset(BigQuery bigQuery, DatasetId dataset, @Nullabl
         // A conflict means the dataset already exists (https://cloud.google.com/bigquery/troubleshooting-errors)
         // This most likely means multiple stages in the same pipeline are trying to create the same dataset.
         // Ignore this and move on, since all that matters is that the dataset exists.
-        throw new IOException(errorMessage.get(), e);
+        throw ErrorUtils.getProgramFailureException(
+          new ErrorCategory(ErrorCategory.ErrorCategoryEnum.PLUGIN), "Dataset already exists", errorMessage.get(),
+          ErrorType.UNKNOWN, true, e);
```

Comment:
similar comment here: https://github.com/data-integrations/google-cloud/blob/601f62bd0005255e75bff19db8711a8898358d49/src/main/java/io/cdap/plugin/gcp/common/GCPErrorDetailsProvider.java#L69

Only `409` means `dataset already exists` for others it means the request failed to create dataset.

Reply:
updated !

```java
try {
  bigQuery.create(builder.build());
} catch (BigQueryException e) {
  if (e.getCode() != 409) {
    // A conflict means the dataset already exists (https://cloud.google.com/bigquery/troubleshooting-errors)
    // This most likely means multiple stages in the same pipeline are trying to create the same dataset.
    // Ignore this and move on, since all that matters is that the dataset exists.
    ErrorUtils.ActionErrorPair pair = ErrorUtils.getActionErrorByStatusCode(e.getCode());
    String errorReason = String.format("%s %s %s", e.getCode(), e.getMessage(), pair.getCorrectiveAction());
    throw ErrorUtils.getProgramFailureException(
      new ErrorCategory(ErrorCategory.ErrorCategoryEnum.PLUGIN), errorReason, errorMessage.get(),
      pair.getErrorType(), true, e);
  }
}
```